src/sentry/integrations/github/webhook.py

```diff
@@ -357,16 +357,31 @@ def _handle(
                         date_added=parse_date(commit["timestamp"]).astimezone(timezone.utc),
                     )
                     for fname in commit["added"]:
+                        language = EXTENSION_LANGUAGE_MAP.get(fname.split(".")[-1])
```

Comment:
what happens if the file name doesn't have an extension?

Reply:
the field is set to None

Reply:
what happens if the file name is the name of an extension? e.g. if i have a file called `tsx`, but its filled with nonsense, not typescript?

Reply:
(just thinking maybe we should validate that an extension exists before trying to use `EXTENSION_LANGUAGE_MAP`

Reply:
i see. if you do `"tsx".split(".")[-1]` you get `"tsx"` back which wouldn't be correct